miner/miner.go

```diff
@@ -2,29 +2,98 @@ package miner
 
 import (
 	"math"
+	"sync"
 	"time"
 
 	log "github.com/Sirupsen/logrus"
 	"github.com/ubclaunchpad/cumulus/blockchain"
 	"github.com/ubclaunchpad/cumulus/consensus"
 )
 
+// currentlyMining is a flag to control the miner.
+var currentlyMining bool
+
+// currentlyMiningLock is a read/write lock to change the Mining flag.
+var currentlyMiningLock sync.RWMutex
```

Comment:
Edit: I see you want to make `Mine()` take an interface and a callback as a means of updating progress. I think using a `Miner` type could help with that. The only thing is, I fell like in mining there is no notion of progress since you can never know for sure how close you are to the solution.